keanu-project/src/main/java/io/improbable/keanu/algorithms/variational/optimizer/Variable.java

```diff
@@ -1,10 +1,19 @@
 package io.improbable.keanu.algorithms.variational.optimizer;
 
-public interface Variable<T> {
+import io.improbable.keanu.network.VariableState;
+import io.improbable.keanu.vertices.Samplable;
+
+public interface Variable<VALUE, STATE extends VariableState> extends Samplable<VALUE> {
 
     VariableReference getReference();
 
-    T getValue();
+    VALUE getValue();
+
+    void setValue(VALUE value);
```

Comment:
NB: this is called in several places:
* `Proposal#apply` and `Proposal#reject`
* `Leapfrog#stepPosition`
* `NUTS#resetVariableValue`